model/src/main/kotlin/licenses/ResolvedLicense.kt

```diff
@@ -121,19 +121,20 @@ data class ResolvedLicense(
 }
 
 private fun Collection<ResolvedCopyrightFinding>.toResolvedCopyrights(process: Boolean): List<ResolvedCopyright> {
-    val allStatements = map { it.statement }
-    val processedStatements = if (process) {
-        CopyrightStatementsProcessor().process(allStatements).toMap()
-    } else {
-        allStatements.associateBy({ it }, { setOf(it) })
+    if (!process) return map { ResolvedCopyright(it.statement, setOf(it)) }
+
+    val statementToFinding = associateBy { it.statement }
+    val (processedStatements, unprocessedStatements) = CopyrightStatementsProcessor().process(statementToFinding.keys)
```

Comment:
I'm not a fan of deconstructing data classes like this. Mainly because it makes assumptions about the property order which should be an implementation details of the class, but also because it has the risk that chosen names are misleading (I think refactoring the names of the data class would also not update them here).